astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -162,16 +170,24 @@ def cirs_to_gcrs(cirs_coo, gcrs_frame):
     # compute the pmatrix, and then multiply by its transpose
     pmat = gcrs_to_cirs_mat(cirs_coo.obstime)
     newrepr = cirs_coo.cartesian.transform(matrix_transpose(pmat))
-    gcrs = GCRS(newrepr, obstime=cirs_coo.obstime)
-
-    # now do any needed offsets (no-op if same obstime and 0 pos/vel)
+    # can't use location.get_gcrs_posvel as that would trigger infinite recursion
+    obscirsloc, obscirsvel = cirs_coo.location.get_posvel(cirs_coo.obstime, CIRS)
```

Comment:
I'm not quite sure what you mean. It doesn't stop the recursion, which arises because if I try to transform this to GCRS the astropy machinery, then `cirs_to_gcrs` is called inside `cirs_to_gcrs` forever

Reply:
I mean doing: 

```
obsgeoloc, obsgeovel = cirs_coo.location.get_posvel(cirs_coo.obstime, CIRS, transform=matrix_transpose(pmat))
```

Reply:
Ah ok. I’m not a fan of that, as I think it’s a confusing api - one request a position in one frame explicitly, but the transform argument means you get a result in another frame. 
 
Anyway, this is made redundant by @mhvk’s PR to this PR